### ISSUE TITLE
Lint `ref_option` ignores methods on public structs

### ISSUE THREAD

**Issue by orium** [rust-lang/rust-clippy#15833]
### Summary

The `ref_option` lint ignores methods on public structs.

### Reproducer

Reproduction:

```rust
struct Foo {  }

impl Foo {
    pub fn bar(&self) -> &Option<u32> {
        todo!()
    }
}
```

if you run `cargo clippy -- -W clippy::ref_option` you get

```
⋮
warning: it is more idiomatic to use `Option<&T>` instead of `&Option<T>`
⋮
```

however, if you make `Foo` public clippy stops emitting this warning.

### Version

```text
clippy 0.1.90 (1159e78c47 2025-09-14)
```

**Comment by ada4a** [comment]
That's not a bug -- Clippy lints generally avoid suggesting to break public API. This is configurable though, see [`avoid-breaking-exported-api`](https://doc.rust-lang.org/clippy/lint_configuration.html#avoid-breaking-exported-api)

**Comment by orium** [comment]
Ok. I find it surprising: I was expecting clippy to be conservative and warn about everything. Up to people to disable lints locally or globally.

**Comment by y21** [comment]
> Up to people to disable lints locally or globally.

It's rather the opposite: clippy should have good defaults out of the box that work for everyone, and you can opt in for more strict behavior through configurations and pedantic/restriction lints

**Comment by ada4a** [comment]
I see your point. Maybe Clippy could emit a separate list of warnings that are silenced because of a configuration option, or do emit the lint but with note about the API breakage, or something -- but all of these will likely annoy some other group of people (e.g. library maintainers, which wouldn't be to really do anything about some piece of API being unidiomatic)
